src/db_impl.cc

```diff
@@ -762,6 +763,80 @@ void TitanDBImpl::ReleaseSnapshot(const Snapshot* snapshot) {
   db_->ReleaseSnapshot(snapshot);
 }
 
+Status TitanDBImpl::DisableFileDeletions() {
+  // Disable base DB file deletions.
+  Status s = db_impl_->DisableFileDeletions();
+  if (!s.ok()) {
+    return s;
+  }
+
+  int count = 0;
+  {
+    // Hold delete_titandb_file_mutex_ to make sure no
+    // PurgeObsoleteFiles job is running.
+    MutexLock l(&delete_titandb_file_mutex_);
+    count = ++disable_titandb_file_deletions_;
```

Comment:
can we use atomic value instead?

Reply:
NO, just like the comment said. If we use atomic value, there is no way to **prevent** `++disable_titandb_file_deletions_` while `TitanDBImpl::PurgeObsoleteFilesImpl` function is running.